### PR TITLE
feat!: split relay configuration between production and staging

### LINKED ISSUE #2420: test: convert tests to use new test relay servers instead of prod servers

### PR DESCRIPTION
**Pull request by dignifiedquire**
- [x] regular tests
- [x] netsim (nothing to do, it uses its own local relays)
- [ ] ~~CLI integration tests~~

## Breaking Changes

- `iroh_net::defaults` is now split into `prod` and `staging`
- added `iroh_net::RelayMode::Staging`
- `iroh_net::discovery::dns:: N0_DNS_NODE_ORIGIN` is now `N0_DNS_NODE_ORIGIN_PROD` and there is now `N0_DNS_NODE_ORIGIN_STAGING`

Closes #2420

### CODE REVIEW

**Review by flub — APPROVED** [review]
Love it when I do a review and than don't hit submit...

LGTM I think, can't immediately think of something better.

The main consequence is that anyone else their test suite will also use the staging servers.  That means we have to treat them as as important as the main ones.  But otherwise probably the desired effect?

File: iroh-net/src/relay/map.rs [hunk 1]
```diff
@@ -14,8 +14,10 @@ use super::RelayUrl;
 pub enum RelayMode {
     /// Disable relay servers completely.
     Disabled,
-    /// Use the default relay map, with relay servers from n0.
-    Default,
+    /// Use the default relay map, with production relay servers from n0.
+    DefaultProd,
+    /// Use the default relay map, with staging relay servers from n0.
+    DefaultStaging,
```
[the remark this answers is not in the document]
Comment (dignifiedquire) [reply iroh-net/src/relay/map.rs:20]: @ramfox any thoughts on which name to pick here?

[the remark this answers is not in the document]
Comment (matheus23) [reply iroh-net/src/relay/map.rs:20]: Just throwing in `Default` & `Staging` into the bikeshed. It's weird to have multiple "default".

[the remark this answers is not in the document]
Comment (dignifiedquire) [reply iroh-net/src/relay/map.rs]: I like it

**Pull request merged**